pilz_trajectory_generation/test/unittest_trajectory_functions.cpp

```diff
@@ -417,6 +426,107 @@ TEST_P(TrajectoryFunctionsTest, testComputePoseIKInvalidFrameId)
                                    false));
 }
 
+/**
+ * @brief Test if self collision is checked with a valid pose and from a starting position that results in first ik result to have self collision.
+ */
+TEST_P(TrajectoryFunctionsTestOnlyGripper, testComputePoseIKSelfCollisionForValidPosition)
+{
+  const std::string frame_id = robot_model_->getModelFrame();
+  const robot_model::JointModelGroup* jmg = robot_model_->getJointModelGroup(planning_group_);
+
+  // create seed
+  std::vector<double> ik_seed_states = {-0.553, 0.956, 1.758, 0.146, -1.059, 1.247};
+  auto joint_names = jmg->getActiveJointModelNames();
+
+  std::map<std::string, double> ik_seed;
+  for (unsigned int i=0; i < ik_seed_states.size(); ++i)
+  {
+    ik_seed[joint_names[i]] = ik_seed_states[i];
+  }
+
+
+  // create expected pose
+  geometry_msgs::Pose pose;
+  pose.position.x = -0.454;
+  pose.position.y = -0.15;
+  pose.position.z = 0.431;
+  pose.orientation.y = 0.991562;
+  pose.orientation.w = -0.1296328;
+  Eigen::Affine3d pose_expect;
+  tf::poseMsgToEigen(pose, pose_expect);
+
+  // compute the ik with one atemt and expect it to self collide and thus fail.
+  std::map<std::string, double> ik_actual1;
+  EXPECT_TRUE(pilz::computePoseIK(robot_model_,
+                                  planning_group_,
+                                  tcp_link_,
+                                  pose_expect,
+                                  frame_id,
+                                  ik_seed,
+                                  ik_actual1,
+                                  false,
+                                  1));
```

Comment:
Having a closer look this seems to rely on odd behaviour of the IK plugin. There was an upstream [PR](https://github.com/ros-planning/moveit/pull/585) and I think, we should re-generate the ikfast_plugin from the current template there. Then the collision-free solution should be delivered on the first attempt (without involving random seeds making the number of calls non-deterministic).

Could you think of another way to implement this "fail without collision check"-test?